File: app/src/main/java/com/carassistant/model/bus/MessageEventBus.java
package com.carassistant.model.bus;

import io.reactivex.Observable;
import io.reactivex.subjects.PublishSubject;

/**
 * Created by AlexLampa on 28.06.2019.
 */
public enum MessageEventBus {

    INSTANCE;

    private PublishSubject<EventModel> bus = PublishSubject.create();

    public void send(EventModel event) {
        bus.onNext(event);
    }

    public Observable<EventModel> toObservable() {
        return bus;
    }

}

File: app/src/main/java/com/carassistant/model/bus/MessageEventBus.kt
package com.carassistant.model.bus

import io.reactivex.Observable
import io.reactivex.subjects.PublishSubject

enum class MessageEventBus {

    INSTANCE;

    private val bus = PublishSubject.create<EventModel>()

    fun send(event: EventModel) {
        bus.onNext(event)
    }

    fun toObservable(): Observable<EventModel> {
        return bus
    }

}